In [1]:
# import statements
import numpy as np
import pandas as pd

In [2]:
# 1. Load indicated columns of csv file into a dataframe, located at https://raw.githubusercontent.com/ryurko/nflscrapR-data/master/play_by_play_data/post_season/post_pbp_2009.csv
df = pd.read_csv("https://raw.githubusercontent.com/ryurko/nflscrapR-data/master/play_by_play_data/post_season/post_pbp_2009.csv", usecols=["play_id", "game_id", "home_team", "away_team", "game_date", "play_type", "replay_or_challenge", "replay_or_challenge_result"])
df

play_id     game_id home_team away_team   game_date play_type  \
0          36  2010010901       CIN       NYJ  2010-01-09   kickoff   
1          61  2010010901       CIN       NYJ  2010-01-09       run   
2          82  2010010901       CIN       NYJ  2010-01-09      pass   
3         104  2010010901       CIN       NYJ  2010-01-09      pass   
4         166  2010010901       CIN       NYJ  2010-01-09       run   
...       ...         ...       ...       ...         ...       ...   
2097     3677  2010020700       IND        NO  2010-02-07      pass   
2098     3700  2010020700       IND        NO  2010-02-07       run   
2099     3721  2010020700       IND        NO  2010-02-07      pass   
2100     3743  2010020700       IND        NO  2010-02-07  qb_kneel   
2101     3768  2010020700       IND        NO  2010-02-07       NaN   

      replay_or_challenge replay_or_challenge_result  
0                       0                        NaN  
1                       0                        NaN  
2                       0                        NaN  
3                       1                     upheld  
4                       0                        NaN  
...                   ...                        ...  
2097                    0                        NaN  
2098                    0                        NaN  
2099                    0                        NaN  
2100                    0                        NaN  
2101                    0                        NaN  

[2102 rows x 8 columns]

In [3]:
# 2. Count number of missing entries in this dataframe.
df.isna().sum().sum()

2157

There are 2157 missing entries in this dataframe.

In [4]:
# 3. Percentage of entries missing.
percent_missing = df.isnull().sum() * 100 / len(df)
percent_missing

play_id                        0.000000
game_id                        0.000000
home_team                      0.000000
away_team                      0.000000
game_date                      0.000000
play_type                      3.330162
replay_or_challenge            0.000000
replay_or_challenge_result    99.286394
dtype: float64

3.33% of entries are missing from play_type, and 99.29% of entries are missing from replay_or_challenge_result

In [5]:
# 4. If you removed all the rows containing a missing value, how many rows would remain?
len(df.dropna())

15

15 rows would remain if one removed all the rows containing a missing value.

In [6]:
# 5. If you removed all the columns containing a missing value, how many columns would remain?
len(df.dropna(axis=1).columns)

6

6 columns would remain if you dropped all columns containing a missing value.

In [7]:
# 6. Repeat Step 1, if your dataframe has changed. Replace all NaN values in the play_type column with ‘pass’.
df = pd.read_csv("https://raw.githubusercontent.com/ryurko/nflscrapR-data/master/play_by_play_data/post_season/post_pbp_2009.csv", usecols=["play_id", "game_id", "home_team", "away_team", "game_date", "play_type", "replay_or_challenge", "replay_or_challenge_result"])
df.fillna({"play_type": "pass"}, inplace=True)
df

play_id     game_id home_team away_team   game_date play_type  \
0          36  2010010901       CIN       NYJ  2010-01-09   kickoff   
1          61  2010010901       CIN       NYJ  2010-01-09       run   
2          82  2010010901       CIN       NYJ  2010-01-09      pass   
3         104  2010010901       CIN       NYJ  2010-01-09      pass   
4         166  2010010901       CIN       NYJ  2010-01-09       run   
...       ...         ...       ...       ...         ...       ...   
2097     3677  2010020700       IND        NO  2010-02-07      pass   
2098     3700  2010020700       IND        NO  2010-02-07       run   
2099     3721  2010020700       IND        NO  2010-02-07      pass   
2100     3743  2010020700       IND        NO  2010-02-07  qb_kneel   
2101     3768  2010020700       IND        NO  2010-02-07      pass   

      replay_or_challenge replay_or_challenge_result  
0                       0                        NaN  
1                       0                        NaN  
2                       0                        NaN  
3                       1                     upheld  
4                       0                        NaN  
...                   ...                        ...  
2097                    0                        NaN  
2098                    0                        NaN  
2099                    0                        NaN  
2100                    0                        NaN  
2101                    0                        NaN  

[2102 rows x 8 columns]

In [8]:
# 7. Make all the values in play_type column UPPER CASE.
df["play_type"] = df["play_type"].str.upper()
df

play_id     game_id home_team away_team   game_date play_type  \
0          36  2010010901       CIN       NYJ  2010-01-09   KICKOFF   
1          61  2010010901       CIN       NYJ  2010-01-09       RUN   
2          82  2010010901       CIN       NYJ  2010-01-09      PASS   
3         104  2010010901       CIN       NYJ  2010-01-09      PASS   
4         166  2010010901       CIN       NYJ  2010-01-09       RUN   
...       ...         ...       ...       ...         ...       ...   
2097     3677  2010020700       IND        NO  2010-02-07      PASS   
2098     3700  2010020700       IND        NO  2010-02-07       RUN   
2099     3721  2010020700       IND        NO  2010-02-07      PASS   
2100     3743  2010020700       IND        NO  2010-02-07  QB_KNEEL   
2101     3768  2010020700       IND        NO  2010-02-07      PASS   

      replay_or_challenge replay_or_challenge_result  
0                       0                        NaN  
1                       0                        NaN  
2                       0                        NaN  
3                       1                     upheld  
4                       0                        NaN  
...                   ...                        ...  
2097                    0                        NaN  
2098                    0                        NaN  
2099                    0                        NaN  
2100                    0                        NaN  
2101                    0                        NaN  

[2102 rows x 8 columns]

In [9]:
# 8. How many distinct play types exist in your data?
len(df["play_type"].unique())

9

9 unique play types exist in my data.

In [10]:
# 9. Make only the ‘home_team’ column name UPPER CASE.
df.rename(columns = {"home_team": "HOME_TEAM"}, inplace=True)

In [11]:
# 10. Randomly sample 8 entries from the dataframe. 
df.sample(n=8)

play_id     game_id HOME_TEAM away_team   game_date play_type  \
689      4032  2010011000       ARI        GB  2010-01-10      PASS   
1086      927  2010011700       MIN       DAL  2010-01-17      PASS   
419      1544  2010011001        NE       BAL  2010-01-10       RUN   
585      1598  2010011000       ARI        GB  2010-01-10       RUN   
1588      338  2010012401        NO       MIN  2010-01-24      PASS   
1070      567  2010011700       MIN       DAL  2010-01-17      PASS   
172       135  2010010900       DAL       PHI  2010-01-09      PASS   
803      2321  2010011601        NO       ARI  2010-01-16      PASS   

      replay_or_challenge replay_or_challenge_result  
689                     0                        NaN  
1086                    0                        NaN  
419                     0                        NaN  
585                     0                        NaN  
1588                    0                        NaN  
1070                    0                        NaN  
172                     0                        NaN  
803                     0                        NaN